server/aesdsocket.c

```diff
@@ -304,76 +475,131 @@ static void handle_client_connection(int client_fd)
     inet_ntop(AF_INET, &(client_addr.sin_addr), client_ip, MAX_IP_LEN);
 
     syslog(LOG_INFO, "Accepted connection from %s", client_ip);
-
-    char *buffer = (char *)malloc(sizeof(char) * BUFFER_SIZE);   //buffer used to read and write data to and from the client
-    if (buffer == NULL) 
-    {
-        syslog(LOG_ERR, "Could not allocate memory");
-        exit(ERROR_MEMORY_ALLOC);
-    } else 
-    {
-        syslog(LOG_ERR, "Successfully created buffer\n");
-    }
-
-    //opening a data file (DATA_FILE) in write-only mode with options to create the file if it doesn't exist and to append data to it.
-    int data_fd = open(DATA_FILE, O_WRONLY | O_CREAT | O_APPEND, FILE_PERMISSIONS);
-    if (data_fd == -1) 
+    
+    //Locking thread mutex
+    if (pthread_mutex_lock(thread_data->mutex) == -1)
```

Comment:
@vidhyapalaniappan this works but the scope of the `thread_data->mutex` lock is quite large, and essentially means only one socket thread can operate at a given time, essentially removes the benefit of threading.